ic-auction/src/lib.rs

```diff
@@ -0,0 +1,8 @@
+pub mod api;
+mod state;
+mod types;
+
+pub mod error;
+
+pub use self::state::*;
+pub use self::types::*;
```

Comment:
1. Don't call the module types, it's entirely nonsense and I don't know why we keep insisting on doing it
2. Don't glob imports, especially since there are only two types in there
3. Don't call module types (this is worth mentioning twice). This is like naming a variable `variable`

Reply:
1. How else you suggest to name it? This is basically a module for domain specific types 🤷 
2. Well, I'll export the module then

Reply:
Isn't every module a module for domain specific types in almost every programming project ever?!?!

Can't they live in the same module as state?

I can't stress how poor it is to name something `types`. If we are making a programming language, sure, but just calling a module "types" tells you nothing about what's in there

Reply:
Take is20 for example
https://github.com/infinity-swap/IS20/blob/main/src/token/api/src/types.rs
Would you really like increasing `state.rs` module by additional 2 hundred lines with structs that are used not only in state? I don't really understand what's the problem with having this module though as I never had this confusion myself (probably because of my haskell background).